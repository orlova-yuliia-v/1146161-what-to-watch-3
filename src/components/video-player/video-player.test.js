import React from "react";
import renderer from "react-test-renderer";
import VideoPlayer from "./video-player.jsx";

const film = {
  poster: `1.jpg`,
  previewUrl: `https://preview-url.com/1.mp4`
};

const {poster, previewUrl} = film;

it(`should render correctly`, () => {
  const tree = renderer.create(<VideoPlayer
    poster={poster}
    previewUrl={previewUrl}
    isPlaying={true}
  />,
  {
    createNodeMock: (element) => {
      if (element.type === `video`) {
        return {
          play: () => {}
        };
      }
      return null;
    }
  })
.toJSON();

  expect(tree).toMatchSnapshot();
});
