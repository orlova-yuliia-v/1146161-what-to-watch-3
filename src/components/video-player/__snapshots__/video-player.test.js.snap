// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<video
  height="175"
  poster="1.jpg"
  src="https://preview-url.com/1.mp4"
  width="280"
/>
`;
